### DISCUSSION TITLE
NX Release package.json out of sync

### DISCUSSION THREAD

**Discussion by atomicrobokid** [nrwl/nx#27385] [Questions]
Hi,

Bit of a weird one,

I have just started using `nx release` in fixed mode to keep packages in sync.

I have the following config in `nx.json`

`"preVersionCommand": "nx run-many --target=build "` - this builds the libs prior to versioning and puts the output in the workspace root `/dist/{libname}` folder.

However, when it then versions, it versions the `package.json` file in the relevant `lib` directory and doesn't update the ones in `/dist/{libname}` so they are always one version behind when `nx release` publishes.

Maybe i'm missing something obvious ❓ 🤷‍♂️ 

Thanks in advance!

**Comment by atomicrobokid** [comment]
The flip side of that is if i set 
```json
"version": {
     "generatorOptions": {
        "packageRoot": "dist/{projectRoot}"
      }
}
```

It bumps the /dist folders package.json, but not the source lib, so they end up out of sync but in a different way 🤷‍♂️

**Comment by MichaelBorde** [comment]
I have the exact same question and use a workaround below.

I do not configure `packageRoot` for version subcommand.
This means nx will alter `package.json` in production code directly.

I configure the publish subcommand `packageRoot` to use `dist` folders.

I execute following commands during a release:

1. `npx nx release version <specifier>`
2. `npx nx run-many -t build`
3. `npx nx release publish`
4. `git push && git push --tags`

Step 2 will build artefacts with updated `package.json` from step 1.
Step 4 seems needed because nx won't push anything out of a changelog subcommand, which is odd.
